src/main/java/com/bobocode/Dialect.java

```diff
@@ -1,5 +1,26 @@
 package com.bobocode;
 
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public interface Dialect {
+    String SELECT_ALL_TEMPLATE = "select * from %s";
+
+    String SELECT_ALL_ID_TEMPLATE = "select * from %s where id = ?";
+
+    String SELECT_ALL_BY_PROPERTIES_TEMPLATE = "select * from %s where %s";
+
+    String AND_WITH_SPACES = " AND ";
+
+    String NAME_EQUALS_VALUE_TEMPLATE = "%s = ?";
+
+    static String prepareWhereClause(Map<String, Object> properties) {
+        return properties.keySet().stream()
+                .sorted()
+                .map(NAME_EQUALS_VALUE_TEMPLATE::formatted)
+                .collect(Collectors.joining(AND_WITH_SPACES));
+    }
+
+    String getLimitOffsetClause(Long limit, Long offset);
```

Comment:
Long will be too big for limit and for offset, will be better to use primitive int.

Reply:
fixed